### PR TITLE
chore: update tokio for tests and transforms

### PR DESCRIPTION
**Pull request by fanatid**
Ref. #2945 

This PR update tokio in `test_util` and transforms (only one tranform use old tokio: `aws_ec2_metadata`).

### CODE REVIEW

**Comment by fanatid** [comment]
Blocked by #3355

File: benches/http.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ fn benchmark_http_no_compression(c: &mut Criterion) {
                 let mut rt = runtime::Runtime::new().unwrap();
 
                 let (topology, _crash) = rt.block_on_std(topology::start(config, false)).unwrap();
-                wait_for_tcp(in_addr);
+                rt.block_on_std(async move { wait_for_tcp(in_addr).await });
```
Comment (bruceg) [inline benches/http.rs:54]: I take it there's no async in this benchmark.

File: src/transforms/aws_ec2_metadata.rs [hunk 2]
```diff
@@ -143,7 +141,7 @@ impl TransformConfig for Ec2Metadata {
         let refresh_interval = self
             .refresh_interval_secs
             .map(Duration::from_secs)
-            .unwrap_or(Duration::from_secs(10));
+            .unwrap_or_else(|| Duration::from_secs(10));
```
Comment (bruceg) [inline src/transforms/aws_ec2_metadata.rs:144]: I'm curious why this is needed. I presume it's a clippy lint. `Duration::from_secs` is `const` and `Duration` implements `Copy` according to the tokio docs, so a closure should be just extra work.

Comment (fanatid) [reply src/transforms/aws_ec2_metadata.rs]: Indeed, `Duration::from_secs` is `const` fn. Thanks @bruceg.

Comment (MOZGIII) [reply src/transforms/aws_ec2_metadata.rs]: https://godbolt.org/z/csrc88
https://godbolt.org/z/vocav6

I'd like to point out that the generated code is effectively the same with optimization, and without closures are still better:

https://godbolt.org/z/43dhh9
https://godbolt.org/z/x4b61n

With closures, the default `Duration` gets created lazily.

Thus, I don't think it's worth bothering to change `unwrap_or_else` to `unwrap_or` for const defaults in general or prefer it for that matter.

**Pull request merged**